### PR TITLE
Enable Travis CI caching of go toolchain artifacts

### PR DESCRIPTION
**Pull request by bobvawter**
I found that I could cut my build times in half on another golang/Travis project by allowing go's caches to be reused across builds.

### CODE REVIEW

**Review by petermattis — APPROVED** [review]
LGTM if this works.

**Comment by petermattis** [comment]
Is there anyway to tell if this works?

**Comment by bobvawter** [comment]
I'll wait for the CI build to run, record the times, and run it again.

Also, the queuing times on `travis-ci.com` are almost non-existent versus running on `.org`.  OSS projects are still free.

**Comment by petermattis** [comment]
> Also, the queuing times on travis-ci.com are almost non-existent versus running on .org. OSS projects are still free.

How do we change to using travis-ci.com?

**Comment by bobvawter** [comment]
https://docs.travis-ci.com/user/migrate/open-source-repository-migration

**Comment by bobvawter** [comment]
The time for a re-build dropped across the board.  The first build on master will create a new cache and then subsequent PRs will be able to take advantage of it.

Initial

Rebuild w/ cache

**Comment by petermattis** [comment]
Nifty trick. Thanks @bobvawter.

PS I attempted to migrate to travis-ci.com following their directions, but ended up in some loops. I'll try again another time.

**Pull request merged**
